pkg/utils/secrets/basic_auth.go

```diff
@@ -73,6 +75,45 @@ func (s *BasicAuthSecretConfig) Generate() (Interface, error) {
 	return s.GenerateBasicAuth()
 }
 
+// GenerateInfoData implements ConfigInterface.
+func (s *BasicAuthSecretConfig) GenerateInfoData() (infodata.InfoData, error) {
+	password, err := utils.GenerateRandomString(s.PasswordLength)
```

Comment:
Hm, I'm a little confused. Now we have multiple functions generating a password, and also multiple functions that can load an existing basic auth data. Is it possible to harmonize this to make it easier for a consumer to know what to call when?

Reply:
I also personally find the division of responsibilities between `SecretsManager`, `SecretsLoader`, `SecretsGenerator`, and `SecretsDeployer` a bit confusing. I think this is the root cause for some of the other comments and might cause issues in the future as well. I think here is no very good way to do such a fine-granular division in this case. I would prefer combining all these 4 types into a single one (e.g. `SecretsManager`) having all `Load`, `Generate`, and `Deploy` methods. It would turn out a bit biggish, but I see this as the "lesser evil" in this case.

Reply:
@rfranzke I think these functions are indeed used in different situations, so perhaps they should stay different but maybe the documentation comments when each of them is used should be added?